source/module_hsolver/diago_david.cpp

```diff
@@ -17,22 +15,30 @@
 
 using namespace hsolver;
 
-template <typename T, typename Device> DiagoDavid<T, Device>::DiagoDavid(const Real* precondition_in)
+
+template <typename T, typename Device>
+DiagoDavid<T, Device>::DiagoDavid(const Real* precondition_in, 
+                                  int diago_david_ndim_in,
```

Comment:
the variable name 'diago_david_ndim_in' is a bit long, how about change it to "david_ndim_in"？